src/client/v2/algod/models/types.ts

```diff
@@ -2738,8 +2754,14 @@ export class DryrunTxnResult extends BaseModel {
     this.disassembly = disassembly;
     this.appCallMessages = appCallMessages;
     this.appCallTrace = appCallTrace;
-    this.budgetAdded = budgetAdded;
-    this.budgetConsumed = budgetConsumed;
+    this.budgetAdded =
+      typeof budgetAdded === 'undefined'
+        ? undefined
+        : ensureSafeInteger(budgetAdded);
```

Comment:
This is a repeat of the previous question (why not coerce to 0). But more broadly, could `ensureSafeInteger` accept `undefined` and return 0?

Reply:
`ensureSafeInteger(undefined)` will error. If you wanted it to coerce undefined to 0, you could do so with `ensureSafeInteger(x ?? 0)` (the `??` being [the null coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing), which is relatively new). The `Transaction` constructor uses this pattern a lot